### ISSUE TITLE
Questionable behavior of DetectPreviewFeatures analyzers

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#5850]
https://github.com/dotnet/roslyn-analyzers/blob/60eb13c44c366205e5a4dec195404477009c41de/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureTests.Generics.cs#L13-L45

Diagnostic span for this test is very inconsistent with it everywhere else.

@jeffhandley Can you confirm if this behavior is correct?

**Comment by Youssef1313** [comment]
Also I'm wondering if this should use `UsesPreviewTypeParameterRuleWithCustomMessage` rule instead of `GeneralPreviewFeatureAttributeRuleWithCustomMessage`.

**Comment by buyaa-n** [comment]
> Diagnostic span for this test is very inconsistent with it everywhere else.
> 
> @jeffhandley Can you confirm if this behavior is correct?

Good catch, that behavior is not correct, I guess that is a leftover from the [work updating diagnostic locations](https://github.com/dotnet/roslyn-analyzers/pull/5402), the span should be around type parameter 

> Also I'm wondering if this should use UsesPreviewTypeParameterRuleWithCustomMessage rule instead of GeneralPreviewFeatureAttributeRuleWithCustomMessage.

Yes, when the diagnostic span moved to the type parameter `UsesPreviewTypeParameterRuleWithCustomMessage` rule should be used

**Comment by Youssef1313** [comment]
@buyaa-n Thanks for confirming. I'm currently working on a re-write for the analyzer. Once I finish it, I'll need to make sure there is no performance regression, then I'll tag you and @jeffhandley for review.
